internal/controllers/s3bucket/s3_agent.go

```diff
@@ -0,0 +1,95 @@
+package s3bucket
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	s3v1alpha1 "github.com/snapp-incubator/s3-operator/api/v1alpha1"
+	"github.com/snapp-incubator/s3-operator/pkg/consts"
+)
+
+// S3Agent wraps the s3.S3 structure to allow for wrapper methods
+type S3Agent struct {
+	Client *s3.S3
+}
+
+func newS3Agent(accessKey, secretKey, endpoint string, debug bool) (*S3Agent, error) {
```

Comment:
There's a similar function in `s3userclaim_test.go`. Could we please put this s3_agent code in a higher-level directory and reuse its code?

Reply:
Sure thing. Do you have any specific path in mind?

Reply:
`internal/s3_agent` for example 😄

Reply:
Done.